core/message_queue_test.go

```diff
@@ -0,0 +1,291 @@
+package core_test
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/core"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+var keys = types.MustGenerateKeyInfo(3, types.GenerateKeyInfoSeed())
+var addresses = make([]address.Address, len(keys))
+var signer = types.NewMockSigner(keys)
+var msgSeq = 0
+var gasPrice = *types.ZeroAttoFIL
+var gasUnits = types.NewGasUnits(0)
+
```

Comment:
Can we avoid having global variables in tests? Other testing functions will share this space. I'd prefer setting this stuff inside a test suite, in helper functions from within tests.  It has caused me headaches in places like default_syncer_test.go. It is a lot like RSpec before(:all) which also can have unexpected side effects, but there  it's at least confined to the file itself.

Reply:
Thank you, I was looking for a pattern here and couldn't find a clear one. I will do that.

The minor reason I chose top-level vars was so I could define helper functions below the test code. If I move them inside the test method I think I have to declare the helpers before the test code, as `func` vars.